lti_consumer/plugin/views.py

```diff
@@ -177,50 +175,60 @@ def launch_gate_endpoint(request, suffix=None):  # pylint: disable=unused-argume
 
     context = {}
 
-    course_key = usage_key.course_key
-    course = compat.get_course_by_id(course_key)
-    user_role = compat.get_user_role(request.user, course_key)
-    external_user_id = compat.get_external_id_for_user(request.user)
-    lti_consumer = lti_config.get_lti_consumer()
-
     try:
-        # Pass user data
-        # Pass django user role to library
-        lti_consumer.set_user_data(user_id=external_user_id, role=user_role)
-
-        # Set launch context
-        # Hardcoded for now, but we need to translate from
-        # self.launch_target to one of the LTI compliant names,
-        # either `iframe`, `frame` or `window`
-        # This is optional though
-        lti_consumer.set_launch_presentation_claim('iframe')
-
-        # Set context claim
-        # This is optional
-        context_title = " - ".join([
-            course.display_name_with_default,
-            course.display_org_with_default
-        ])
-        # Course ID is the context ID for the LTI for now. This can be changed to be
-        # more specific in the future for supporting other tools like discussions, etc.
-        lti_consumer.set_context_claim(
-            str(course_key),
-            context_types=[LTI_1P3_CONTEXT_TYPE.course_offering],
-            context_title=context_title,
-            context_label=str(course_key)
+        lti_consumer = lti_config.get_lti_consumer()
+        # Decode the JWT and verify the signature before proceeding to verify that the lti_message_hint
+        # JWT came from the platform.
+        lti_message_hint_data = lti_consumer.decode_token(lti_message_hint)
```

Comment:
I think technically the JWT key handling stuff knows which part of the private key to use when verifying the signature because they JWT carries a key ID field that says "signed with this key". We have both the public and private parts of the key available so when the JWT library knows it was signed with key foo (private side) it knows to verify with key foo (public side)